### PR TITLE
feat: integrate patch api into nameeditiondrawer

### LINKED ISSUE #274: Integrate `PATCH` API into `NameEditionDrawer`
## Description

Integrate the existing `PATCH /api/sharehouse/:share_house_id` and `PATCH /api/category/:category_id` into the [`NameEditionDrawer`](https://github.com/Tascurator/tascurator-frontend/blob/develop/src/components/ui/drawers/NameEditionDrawer.tsx) component to enable updating share house name or category name. Please ensure that the UI correctly reflects the updates and handles any necessary user interactions with toasts for confirmations and error messages.

## Criteria

- [ ] Modify [`NameEditionDrawer`](https://github.com/Tascurator/tascurator-frontend/blob/develop/src/components/ui/drawers/NameEditionDrawer.tsx) to update names by calling `PATCH /api/sharehouse/:share_house_id` or `PATCH /api/category/:category_id` dynamically depending on the type of the drawer
- [ ] Display a success toast when it's completed without errors
- [ ] Display an error toast with an error from the backend when it fails
- [ ] If applicable, delete hard-coded code like to simulate data mutation in the previous development

### PR DESCRIPTION
**Pull request by yuki-92c**
## Overview

- I implemented patch api into name edition drawer.

## Changes

- changed
`src/components/landlord-dashboard/LandlordDashboard.tsx`
`src/components/ui/accordion/AccordionCategoryItem.tsx`
`src/components/ui/drawers/NameEditionDrawer.tsx`
`src/components/ui/header.tsx`

## Review points

- Change Category/ Share house name
- Display a toast

## Screenshots or videos

-

https://github.com/user-attachments/assets/41726f18-2b41-41c9-9caa-22c28982fec1

## Notes

- The following sections need to be fixed.
  - When there are multiple categories/share houses, the modified ones move to the bottom of the list.
  - The header name on the share house detail page is hardcoded (I decided to address this in a separate PR after the implementation of the share house detail page is completed).

## Assignee Checklist:

- [ ] The base branch is correct (no accidental merges)
- [ ] The branch name follows our branch naming rules
- [ ] The PR title follows our PR title rules
- [ ] My code follows our coding style
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings or errors in the console

## Reviewer Checklist:

- [x] Code readability and simplicity
- [x] Follows best practices and coding standards
- [ ] Understandable and maintainable code

### CODE REVIEW

File: src/components/ui/drawers/NameEditionDrawer.tsx [hunk 1]
```diff
@@ -104,34 +110,61 @@ export const NameEditionDrawer = ({
...
     // Submit the form data
     await new Promise((resolve) => setTimeout(resolve, 1000));
 
-    // Update the name based on the id
-    if (name) {
-      setIsLoading(false);
+    try {
+      if (isSharehouse) {
+        console.log('Updating the Sharehouse:', data.name);
+        console.log('Sharehouse ID:', shareHouseId);
```
Comment (akiomatic) [inline src/components/ui/drawers/NameEditionDrawer.tsx]: Could you delete these console logging?

File: src/components/ui/drawers/NameEditionDrawer.tsx [hunk 2]
```diff
@@ -104,34 +110,61 @@ export const NameEditionDrawer = ({
...
+            name: data.name,
+          },
+        });
+        const sharehouseData = await resSharehouse.json();
+        if ('error' in sharehouseData) {
+          throw new Error(sharehouseData.error);
+        }
+      } else {
+        console.log('Updating the Category:', data);
+        console.log('Category ID:', categoryId);
```
Comment (akiomatic) [inline src/components/ui/drawers/NameEditionDrawer.tsx]: Could you delete these console logs?

File: src/components/ui/drawers/NameEditionDrawer.tsx [hunk 3]
```diff
@@ -50,7 +55,6 @@ const getSchema = (type: 'sharehouse' | 'category') => {
  * @param type - The type to determine if editing a 'sharehouse' or 'category'
  *
  * @example
- * const [open, setOpen] = useState(false);
```
Comment (akiomatic) [inline src/components/ui/drawers/NameEditionDrawer.tsx:53]: Could you bring this back?

File: src/components/ui/drawers/NameEditionDrawer.tsx [hunk 4]
```diff
@@ -31,6 +34,8 @@ interface INameEditionDrawer {
   open: boolean;
   setOpen: (value: boolean) => void;
   type: 'sharehouse' | 'category';
+  shareHouseId?: string;
+  categoryId?: string;
```
Comment (akiomatic) [inline src/components/ui/drawers/NameEditionDrawer.tsx]: Could you modify this to accept only one ID, which will change dynamically based on the given `type`? For example, if `type` is `sharehouse`, the ID will be interpreted as `shareHouseId`.

File: src/components/ui/drawers/NameEditionDrawer.tsx [hunk 5]
```diff
@@ -104,34 +110,61 @@ export const NameEditionDrawer = ({
 
   // TODO: Implement the onSubmit click functionality
   const onSubmit: SubmitHandler<FormSchema> = async (data) => {
-    setIsLoading(true);
-    setOpen(true);
-
     // Submit the form data
```
Comment (akiomatic) [inline src/components/ui/drawers/NameEditionDrawer.tsx]: This comment should be removed or moved to the correct position.

File: src/components/ui/drawers/NameEditionDrawer.tsx [hunk 6]
```diff
@@ -104,34 +110,61 @@ export const NameEditionDrawer = ({
 
   // TODO: Implement the onSubmit click functionality
```
Comment (akiomatic) [inline src/components/ui/drawers/NameEditionDrawer.tsx]: Could you remove this TODO?

File: src/components/ui/header.tsx [hunk 7]
```diff
@@ -262,7 +263,7 @@ const Header = ({ type, pageTitle, sharehouseId }: IHeaderContainerProps) => {
         ) : type === 'HeaderItemWithDropDown' ? (
           <HeaderItemWithDropDown
             pageTitle={pageTitle}
-            sharehouseId={sharehouseId!}
+            sharehouseId={sharehouseId ?? ''}
```
Comment (akiomatic) [inline src/components/ui/header.tsx:266]: Ideally, `sharehouseId` should **never be undefined** when the `type` is `HeaderItemWithDropdown`. I understand the issue mentioned in the PR description, so when addressing it, please ensure `sharehouseId` is always defined for `HeaderItemWithDropdown` type.

**Review by akiomatic — CHANGES_REQUESTED** [review]
Thank you. I've left some comments!

**Comment by yuki-92c** [comment]
thanks Akio for reviewing. I modified the code so please check again

**Comment by yuki-92c** [comment]
Thank you Akio. I fixed code so please check again.

File: src/components/ui/drawers/NameEditionDrawer.tsx [hunk 8]
```diff
@@ -31,6 +34,9 @@ interface INameEditionDrawer {
   open: boolean;
   setOpen: (value: boolean) => void;
   type: 'sharehouse' | 'category';
+  id?: string;
+  // shareHouseId?: string;
+  // categoryId?: string;
```
Comment (akiomatic) [inline src/components/ui/drawers/NameEditionDrawer.tsx]: Could you remove the old props without commenting them out?

File: src/components/ui/drawers/NameEditionDrawer.tsx [hunk 9]
```diff
@@ -104,34 +110,57 @@ export const NameEditionDrawer = ({
...
     await new Promise((resolve) => setTimeout(resolve, 1000));
 
-    // Update the name based on the id
-    if (name) {
-      setIsLoading(false);
+    try {
+      if (isSharehouse) {
+        const resSharehouse = await api.sharehouse[':shareHouseId'].$patch({
+          param: {
+            shareHouseId: id!,
```
Comment (akiomatic) [inline src/components/ui/drawers/NameEditionDrawer.tsx]: The non-null assertion could lead to unintended behavior later in development. Could you either add a comment on the line to remind us to fix it later or add an if statement to ensure the submission only happens if the ID is not null?

File: src/components/ui/drawers/NameEditionDrawer.tsx [hunk 10]
```diff
@@ -104,34 +110,57 @@ export const NameEditionDrawer = ({
...
+          },
+        });
+        const sharehouseData = await resSharehouse.json();
+        if ('error' in sharehouseData) {
+          throw new Error(sharehouseData.error);
+        }
+      } else {
+        const resCategory = await api.category[':categoryId'].$patch({
+          param: {
+            categoryId: id!,
```
Comment (akiomatic) [inline src/components/ui/drawers/NameEditionDrawer.tsx]: The non-null assertion could lead to unintended behavior later in development. Could you either add a comment on the line to remind us to fix it later or add an if statement to ensure the submission only happens if the ID is not null?

File: src/components/ui/drawers/NameEditionDrawer.tsx (hunk 6 above)
Comment (akiomatic) [reply src/components/ui/drawers/NameEditionDrawer.tsx]: Could you fix this too?

File: src/components/ui/drawers/NameEditionDrawer.tsx [hunk 11]
```diff
@@ -90,7 +96,7 @@ export const NameEditionDrawer = ({
   const {
     register,
     handleSubmit,
-    formState: { errors, isValid },
+    formState: { errors, isValid, isSubmitting },
```
Comment (akiomatic) [inline src/components/ui/drawers/NameEditionDrawer.tsx:99]: Could you remove `isValid` as you no longer use it?

**Review by akiomatic — CHANGES_REQUESTED** [review]
Thank you. Could you please check my comments and make some modifications?

**Review by akiomatic — APPROVED** [review]
Thank you! LGTM!

**Pull request merged**